### DISCUSSION TITLE
Wake up from Sleep to perform script

### DISCUSSION THREAD

**Discussion by WackoCommander** [koreader/koreader#12060] [Q&A]
Hi, I am a linux beginner and was wondering, if there is a way to run a linux script while the device is in sleep. I want to have a script that pushes the sqlite stats to my mysql server. Thanks.

**Comment by pazos** [comment]
Nope. When the device is slept it can't do actions until it wakes up. Network actions might add some delay as you'll need to wait a few more seconds for network association.

It is better to run that script **before** suspending. That should be quick if your network is currently enabled.

**Reply by WackoCommander** [reply]
Is there a way to trigger the script once I have pressed the sleep button?

**Reply by pazos** [reply]
With an user patch you can hook on `Device:onSuspend` and call your shell script from there.
But there're are better places to hook, like when you return to the FM from a document.
